gatherers/manifest.js

```diff
@@ -61,32 +71,29 @@ class Manifest extends Gather {
      * potentially lead to a different asset. Using the original manifest
      * resource is tracked in issue #83
      */
-    return driver.querySelector('head link[rel="manifest"]')
-      .then(node => {
-        if (!node) {
-          this.artifact = Manifest._errorManifest('No <link rel="manifest"> found in DOM.');
-          return;
-        }
+    return driver.sendCommand('Runtime.evaluate', {
+      expression: getManifestContent,
+      returnByValue: true
+    }).then(returnedData => {
+      if (returnedData.result.value === undefined ||
+          returnedData.result.value === null ||
+          returnedData.result.value === {}) {
+       // The returned object from Runtime.evaluate is an enigma
+       // Sometimes if the returned object is not easily serializable,
+       // it sets value = {}
+        throw new Error('Failed to get proper result from runtime eval');
```

Comment:
Added the name in latest commit
On Fri, Apr 8, 2016 at 9:58 PM Brendan Kenny <notifications@github.com>
wrote:

> In gatherers/manifest.js
> <https://github.com/GoogleChrome/lighthouse/pull/164#discussion_r59108176>
> :
>
> > -      .then(node => {
> > -        if (!node) {
> > -          this.artifact = Manifest._errorManifest('No <link rel="manifest"> found in DOM.');
> > -          return;
> > -        }
> > +    return driver.sendCommand('Runtime.evaluate', {
> > +      expression: getManifestContent,
> > +      returnByValue: true
> > +    }).then(returnedData => {
> > +      if (returnedData.result.value === undefined ||
> > +          returnedData.result.value === null ||
> > +          returnedData.result.value === {}) {
> > +       // The returned object from Runtime.evaluate is an enigma
> > +       // Sometimes if the returned object is not easily serializable,
> > +       // it sets value = {}
> > +        throw new Error('Failed to get proper result from runtime eval');
>
> Yeah, no gather fault reporting yet, though we need it. You could add a
> name or make it so it reports or in the debugString (even though it's not a
> user error, it may be fine for now)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/lighthouse/pull/164/files/d5af34e1847a32a68830d3684ec959299d36543f#r59108176>
>